components/sensor/atm90e32.rst

```diff
@@ -0,0 +1,139 @@
+ATM90E32 Power Sensor
+====================
+
+.. seo::
+    :description: Instructions for setting up ATM90E32 energy metering sensors
+    :image: atm90e32.png
+    :keywords: ATM90E32, CircuitSetup, Split Single Phase Real Time Whole House Energy Meter, Expandable 6 Channel ESP32 Energy Meter Main Board
+
+The ``atm90e32`` sensor platform allows you to use your ATM90E32 voltage/current and power sensors
+(`datasheet <http://ww1.microchip.com/downloads/en/devicedoc/Atmel-46003-SE-M90E32AS-Datasheet.pdf>`__) sensors with
+ESPHome. This sensor is commonly found in CircuitSetup 2 and 6 channel energy meters.
+
+The atm90e32 IC can measure up to three AC voltages although typically only one
+voltage measurement would be used for the mains electricity phase of a
+household. Three current measurements are read via CT clamps. 
+
+The CircuitSetup Split Single Phase Real Time Whole House Energy Meter can read 2 current channels and one voltage channel.
+.. figure:: images/atm90e32-cs-2chan-full.jpg
+    :align: center
+    :width: 50.0%
+
+    CircuitSetup Split Single Phase Real Time Whole House Energy Meter.
+
+The CircuitSetup Expandable 6 Channel ESP32 Energy Meter can read 6 current channels and 2 voltage channels at a time, this board has two atm90e32 ICs and requires two sensors to be configured in ESPHome.
+.. figure:: images/atm90e32-cs-6chan-full.jpg
+    :align: center
+    :width: 50.0%
+
+    CircuitSetup Expandable 6 Channel ESP32 Energy Meter Main Board.
+
+The IC supports a number of measurements, these are not yet exported by ESPHome
+Electrical parameters measurement: Vrms, Irms, mean active/reactive/apparent
+power, frequency, power factor and phase angle.
+Enhanced event detection: sag, over voltage, phase loss, over current, reverse V/I
+phase sequence, calculated neutral line current INC over-current and frequency
+upper and lower threshold.
+
+.. code-block:: yaml
+
+    # Example CS 2-channel entry
+    sensor:
+      - platform: atm90e32
+        cs_pin: 5
+        voltage_a:
+          name: "EMON Line Voltage A"
+        current_a:
+          name: "EMON CT1 Current"
+        current_b:
+          name: "EMON CT2 Current"
+        power:
+          name: "EMON Total Active Power CT1-2"
+        frequency:
+          name: "EMON Line Frequency"
+        line_freq: 50Hz
+        gain_pga: 2X
+        gain_volt_a: 47660
+        gain_ct_a: 12577
+        gain_ct_b: 12577
+        update_interval: 60s
+
+    # Example CS 6-channel entry
```

Comment:
Ok, this example is too long - Can we leave the above as the main configuration example and move this to a separate section below?